support/cas-server-support-aup-rest/src/main/java/org/apereo/cas/aup/RestAcceptableUsagePolicyRepository.java

```diff
@@ -81,21 +82,24 @@ public Optional<AcceptableUsagePolicyTerms> fetchPolicy(final RequestContext req
             val rest = aupProperties.getRest();
             val url = StringUtils.appendIfMissing(rest.getUrl(), "/").concat("policy");
             val principal = WebUtils.getAuthentication(requestContext).getPrincipal();
+            val request = WebUtils.getHttpServletRequestFromExternalWebflowContext(requestContext);
 
             val exec = HttpUtils.HttpExecutionRequest.builder()
                 .basicAuthPassword(rest.getBasicAuthPassword())
                 .basicAuthUsername(rest.getBasicAuthUsername())
                 .method(HttpMethod.valueOf(rest.getMethod().toUpperCase()))
                 .url(url)
-                .parameters(CollectionUtils.wrap("username", principal.getId()))
+                .parameters(CollectionUtils.wrap("username", principal.getId(),
+                    "locale", request.getLocale().toString()))
                 .build();
             response = HttpUtils.execute(exec);
             val statusCode = response.getStatusLine().getStatusCode();
+            val result = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
```

Comment:
What happens if the API returns back a 500 with nothing in the response body? What would this do?

Reply:
Any Exception (my guess would be NPE) in this code would caught by the catch block l103, so cas would log the error and proceed with the return optional l108 => the process would be the same as it is now. The logged error wouldn't reflect the actual error though but it will give a hint that something went wrong during the fetch process. 
I can refactor further if you want

Reply:
In this case, it's not that something went wrong. An API endpoint should be free to report back an error code without having to produce a response body. Body is optional. So the exception that would be produced and logged in this case would just be *noise*. You should consume the body only if the response code is a success. We don't strictly need to log it here on the CAS side, or, there should be a way to consume the body without causing an exception (consuming and catching the error is not an option).

Reply:
Ok i see, part of me realy want to have the response content in the logs (we could add a method in httputils) but another part of me really wonder how long it's going to take considering i have a really shallow knowledge of the apache http api. Long story short, considering that there is a good chance that people using this feature have total control over the api providing the policy text, we could just consume the content when the reponse is 2xx and add a warn log. Something like this : `LOGGER.warn("AUP fetch policy request returned with response code [{}] check your API for problems", statusCode);`
This way we can fix the problem and i can always come back later with a better solution for logging the content (if needed)

Reply:
Yes, the premise with almost all REST-based implementation is that the deployment/org actually owns the API. If they needed to, they could track the failures on their own and figure out what's going on. Logging it on the CAS side generally presents no substantial benefit.